### DISCUSSION TITLE
How to determine which package(s) where installed in afterAllInstalled hook?

### DISCUSSION THREAD

**Discussion by Strandedpirate** [yarnpkg/berry#2495] [Q&A] [answered]
I'm writing a plugin that watches `afterAllInstalled`. How do I know which package(s) where installed using this hook? It seems to give me everything and the kitchen sink but not the actually identity of the package(s) installed.

**Comment by arcanis** [comment, marked as answer]
You might be looking for [those hooks](https://github.com/yarnpkg/berry/blob/master/packages/plugin-essentials/sources/commands/add.ts#L274-L282) instead. They however only trigger on `yarn add` (ie not if you add something directly into your package.json), so YMMV. Apart from that, Yarn doesn't keep track of the previous dependency tree (it always just regenerates it and compare with what's on the disk), so you would have to make your own bookkeeping.

**Reply by Strandedpirate** [reply]
Thanks, that works.
